src/test/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationServiceTest.java

```diff
@@ -417,9 +464,46 @@ void testInvalidEnumTypeThrows() {
             publicationService.findAllBySearch(CaseSearchTerm.valueOf("invalid"), TEST_VALUE, true));
     }
 
+    @Test
+    void testTriggerIfDateIsFuture() throws IOException {
+        try (LogCaptor logCaptor = LogCaptor.forClass(PublicationService.class)) {
+            publicationService.checkAndTriggerSubscriptionManagement(artefactInTheFuture);
+            assertEquals(
+                0,
+                logCaptor.getInfoLogs().size(),
+                "Should not have returned a log as no trigger was sent."
+            );
+        } catch (Exception ex) {
+            throw new IOException(ex.getMessage());
+        }
+    }
+
+    @Test
+    void testTriggerIfDateIsNow() {
+        when(subscriptionManagementService.sendArtefactForSubscription(artefactFromNow)).thenReturn(SUCCESSFUL_TRIGGER);
+        try (LogCaptor logCaptor = LogCaptor.forClass(PublicationService.class)) {
+            publicationService.checkAndTriggerSubscriptionManagement(artefactFromNow);
+            assertEquals(SUCCESSFUL_TRIGGER, logCaptor.getInfoLogs().get(0),
+                         "should have returned the Subscription List."
+            );
+        }
+    }
+
+    @Test
+    void testTriggerIfDateIsPast() {
+        when(subscriptionManagementService.sendArtefactForSubscription(artefactFromThePast)).thenReturn(
+            SUCCESSFUL_TRIGGER);
+        try (LogCaptor logCaptor = LogCaptor.forClass(PublicationService.class)) {
+            publicationService.checkAndTriggerSubscriptionManagement(artefactFromThePast);
+            assertEquals(SUCCESSFUL_TRIGGER, logCaptor.getInfoLogs().get(0),
+                         "Should have returned the subscription list"
+            );
+        }
+    }
+
```

Comment:
looking at the O notation for the logic in pub services, you should have at least 5 tests for your date logic, some obvious missing ones i can see is the null values, and if the display from and to are the same date

Reply:
Potentially sorted....